swap_meet/vendor.py

```diff
@@ -1,2 +1,76 @@
+from pytest import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory= None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+
+        return item
+    
+    def get_by_category(self, category):
+        items_in_category = [item for item in self.inventory 
+                            if item.category == category]
+        return items_in_category 
+
+    def find_index(self, item_to_find):
+        i = 0
+        for item in self.inventory:
+            if item == item_to_find:
+                my_item_index = i
+            i += 1
+        
+        return my_item_index
+    
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+
+        my_item_index = self.find_index(my_item)
+        their_item_index = other.find_index(their_item)
+
+        self.inventory[my_item_index] = their_item
+        other.inventory[their_item_index] = my_item
+
+        return True
+    
+    def swap_first_item(self, other):
+        if not self.inventory or not other.inventory:
+            return False
+
+        my_first_item = self.inventory[0]
+        their_first_item = other.inventory[0]
+
+        return self.swap_items(other, my_first_item, their_first_item)
+    
+    def get_best_by_category(self, category):
+        if not self.inventory:
+            return False
+        
+        best_item_by_cat = self.inventory[0]
```

Comment:
👀 There's no guarantee that this item is in our category, so we could start out with a condition that's too high for any of the desired category to beat (consider the case where the first item is not in our category with a condition of 5, and the best item in our category comes later with a condition of 3).

Consider first filtering the invetory to those only in the category (which might be an empty result), then find the best condition in that filtered list, which we could do explicitly as you do here, or with the help of `max` with a lambda.